### DISCUSSION TITLE
Notification filter not working?

### DISCUSSION THREAD

**Discussion by gizburdt** [spatie/laravel-failed-jobs-monitor#77] [General]
Hi, as always, awesome package!

I was trying to filter out some exceptions in failed jobs to be pushed to Slack. So I was reading the docs and found this:

`'notificationFilter' => [App\Notifications\FailedJobNotification::class, 'notificationFilter'],`

But it didn't work, so I started debugging the package and it looks like the above code won't pass the is_callable check in `FailedJobBotifier::shouldSendNotification`. It will work when I change the config from `App\Notifications\FailedJobNotification::class` to `new App\Notifications\FailedJobNotification`.I think this code is less pretty, so I thought I will start a discussion about it. Do you think the code needs to be fixed or the docs need to be changed?

**Comment by gizburdt** [comment]
`new App\Notifications\FailedJobNotification` will also prevent the cache from being serialized, so it's not an option

**Comment by gizburdt** [comment]
In the test, this is called: `https://github.com/spatie/laravel-failed-job-monitor/blob/main/tests/FailedJobMonitorTest.php#L59`.

This will work because `$this` is an object, so this code will pass is_callable(). But in the real world and in the example, the first argument is a string of a class, so the is_callable won't pass.

**Comment by gizburdt** [comment]
The method needs to be static

**Reply by freekmurze** [reply]
Could you PR a fix and a test that would fail without your fix?

**Reply by gizburdt** [reply]
I tried, but the strange thing is the test will pass in all cases. But when I test it in a project, the only way it will work is when the method is static. Any ideas?

**Reply by brandon-hoffman** [reply]
@ gizburdt we have recently encountered the same issue. Have you been able to find a solution?

**Reply by gizburdt** [reply]
When I use a static method it works
